### DISCUSSION TITLE
Any way to assign a background image to a shape?

### DISCUSSION THREAD

**Discussion by mscofield0** [uxbox/uxbox#1201] [Q&A] [answered]
I'd like to assign an image to a ellipse shape as in like a profile icon thing. I don't see how to do it from the UI.

**Comment by elhombretecla** [comment, marked as answer]
Hi @mscofield0 :)

The best way to do that on Penpot by the moment is using masks.

[https://help.penpot.app/user-guide/layer-basics/#masking-layers](https://help.penpot.app/user-guide/layer-basics/#masking-layers)

**Reply by mscofield0** [reply]
Thank you

**Comment by gingerbeardman** [comment]
Watching this.

My use case: I want to apply 8x8 MacPaint pixel patterns to shapes.

So I think I'd need bitmap image fill...
- that is tilable
- whose tile size defaults to 1:1 (100%) of chosen fill image dimensions
- that scales with nearest neighbour algo when zoomed in
- is easily added to library
- and easy to apply or copy between shapes

I think that's all 😅

- Figma can do it with no problems
- Sketch can do it but at 200% renders the image blurry (but not at 100% or 400%)
- Affinity Designer can do it but the sizing of the chosen fill image is almost uncontrollable

Old school
- Fireworks could do it
- FreeHand could do it
- Xara Designer (Affinity's "predecessor") could do it by setting dimensions of chosen fill image
